src/ellswift.rs

```diff
@@ -291,13 +293,17 @@ impl ElligatorSwiftSharedSecret {
 /// This distinction is important because the different parties compute different
 /// hashes of the shared secret.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[deprecated(since = "0.30.0", note = "Use `Party` instead")]
+#[allow(deprecated)]
```

Comment:
In c83c5a762a0a014acb02e085f17ea12fb6038c6c:

Why do we have `deprecated` and `allow(deprecated)` on the same object?